### ISSUE TITLE
dataset does not render

### ISSUE THREAD

**Issue by GHolk** [cheeriojs/cheerio#2080]
html's dataset attributes (`data-` prefix attributes) do not render when serializing.

```term
> let $anchor = $('<a>').text('my link').attr('href', '..').data('test', 'test value')
undefined
> $anchor.data('test')
'test value'
> $.html($anchor)
'<a href="..">my link</a>'
> $anchor[0].data
{ test: 'test value' }
> $anchor[0].attribs
[Object: null prototype] { href: '..' }
```

but one can still use `.attr()` to access `data-` prefix attribute.

**Comment by GHolk** [comment]
There is a ancient closed issue #283

**Comment by fb55** [comment]
As #283 shows, this is unfortunately the API we inherited from jQuery. You can always use `.attr` to set the attribute values.
